MEASURE.md

```diff
@@ -0,0 +1,236 @@
+
+**scoring**: proportion | ratio | continuous-variable | cohort
+
+**group**: The section of report defining the stats for a specific indicator (may be disaggregated by stratifier). Each group has following components
+- id: Group name/id.
+- population: The calculations (or name of variable in CQL that defines the value) for each of population components i.e. initial-population, denominator, numerator
+- stratifier: The disaggregations for given population numerator. i.e. by age, by month, by education etc 
+- supplementalData: Any extra data or intermediate calculation to be output to final report
+
+
+## MEASURE LIBRARY
+
+```
+library HOUSEHOLDIND01 version '1'
+...
+include "FHIRHelpers" version '4.0.1' called FHIRHelpers
+...
+// The Measure interval has closed boundaries. Read details https://cql.hl7.org/02-authorsguide.html#interval-values
+parameter "Measurement Period" Interval<DateTime>
+context Patient
+...
+define "All Groups": [Group] G where G.type = 'person'
+define "All Group Members": flatten("All Groups" G return (G.member M return M.entity))
+define "Group": "All Groups" G return G.id
+define "Group Member": "All Group Members" G where "Patient Id" = Split(G.reference, '/')[1]
+define "Patients": {Patient}
+
+define "Males": Patient.gender='male'
+define "Females": Patient.gender='female'
+
+define "Age": CalculateAgeInYearsAt(Patient.birthDate, ToDate("Measurement Period".high))
+
+define "Age Stratifier":
+  case
+    when "Age" < 1 then 'P0Y'
+    ....
+  end
+```
+
+The CQL is referenced by url into Measure. The CQL is translated into elm-json and uploaded a Library on server. There are multiple ways to get elm for CQL. (elm-xml is also a valid standard but it is not recognized by android fhir libraries yet)
+
+**CQL to ELM REST Translator**:
+A [elm REST app](https://github.com/cqframework/cql-translation-service/blob/master/README.md) that can be used to run elm microservice and convert CQL via a REST API. 
+
+
+**CQL to ELM JAVA Translator**:
+A [elm java app](https://github.com/cqframework/clinical_quality_language/blob/master/Src/java/READM.md) that can be used to elm translator on files and get an output. Instructions can be found [here](https://github.com/cqframework/cql-execution#to-execute-your-cql)
+
+**Note**: Above approaches output a json elm which then need to be base64 decoded and copied to the [Library](http://hl7.org/fhir/R4/library.html) content as Attachment.
+
+**Fhir-Resource on FhirCore**:
+The [fhir-resources](https://github.com/opensrp/fhir-resources/fhircore-testing/src/test/resources/measure-report/household-members.feature) repository has a testing module which allows you to not only get the complete Library resource to directly save to server but also allows to test the Measure output and make changes on the fly. Check the cucumber tests fhircore-testing/src/test/resources/measure-report/household-members.feature. 
```

Comment:
Can we add a link to the line number or actual file containing the function to where this code is being tested